lib/spack/spack/cmd/create.py

```diff
@@ -184,14 +187,29 @@ class CMakePackageTemplate(PackageTemplate):
     """Provides appropriate overrides for CMake-based packages"""
 
     base_class_name = 'CMakePackage'
-
     body = """\
     def cmake_args(self):
         # FIXME: Add arguments other than
         # FIXME: CMAKE_INSTALL_PREFIX and CMAKE_BUILD_TYPE
         # FIXME: If not needed delete this function
         args = []
         return args"""
+    def __init__(self, name, url, versions):
+        PackageTemplate.__init__(self, name, url, versions)
+        with spack.stage.Stage(url, name) as variant_stage:
+            variant_stage.fetch()
+            variant_stage.expand_archive()
+
+            variant_stage.path
+            with spack.working_dir(variant_stage.source_path + "/build", create=True):
+                my_cmake = which('cmake')
+                my_cmake("..")
```

Comment:
> Is there a more powerful way to do this which knows where the CMakeLists file exists?

`spack create` doesn't choose `CMakePackage` unless it can find `CMakeList.txt` in the root directory of the tarball, so I think you're safe.